pkg/reconciler/buildrun/resources/sources/bundle_core.go

```diff
@@ -241,6 +241,7 @@ func unpack(in io.Reader, targetPath string) error {
 			}
 
 			if _, err := io.Copy(file, tr); err != nil {
+				file.Close()
```

Comment:
I think it is better to add a `defer file.Close()` immediately after the `if err != nil ...` block that precedes the io.Copy call

that way, if later changes get inserted in between these lines, we don't lose the call to close

Reply:
I searched for an alternative to this, too, but since it is in a `for` loop, I think there is no other way to write it. The only other option would be to externalize the loop block into a separate function, but I figured that would be overkill.

Reply:
yeah I've done the externalize in a for loop before .... I'd be inclined to do that / in my opinion at least do not consider it overkill